packages/signers/meta-signers/src/multisig-signer.ts

```diff
@@ -0,0 +1,199 @@
+      }
+
+      // Try as_multi_threshold_1
+      if (multisig.threshold === 1) {
+        try {
+          const { location, codec } = dynamicBuilder.buildCall(
+            "Multisig",
+            "as_multi_threshold_1",
+          )
+          const payload = codec.enc({
+            other_signatories: otherSignatories.map(toSS58),
+            call: dynamicBuilder.buildDefinition(lookup.call).dec(callData),
+          })
+          const wrappedCallData = mergeUint8(new Uint8Array(location), payload)
+          return signer.signTx(
+            wrappedCallData,
+            signedExtensions,
+            metadata,
+            atBlockNumber,
+            hasher,
+          )
+        } catch (_) {}
+      }
+
+      const unsignedExtrinsic = mergeUint8(new Uint8Array([4]), callData)
+      const [multisigInfo, weightInfo] = await Promise.all([
+        getMultisigInfo(toSS58(multisigId), Binary.fromBytes(callHash)),
+        txPaymentInfo(
+          Binary.fromBytes(unsignedExtrinsic),
+          unsignedExtrinsic.length,
+        ),
+      ])
+
+      if (
+        multisigInfo?.approvals.some((approval) => {
+          const info = getSs58AddressInfo(approval)
+          return info.isValid && u8ArrEq(info.publicKey, signer.publicKey)
+        })
+      ) {
+        throw new Error("Multisig call already approved by signer")
+      }
+
+      const method = options.method(
+        multisigInfo?.approvals ?? [],
+        multisig.threshold,
+      )
+
+      let wrappedCallData
+      try {
+        const { location, codec } = dynamicBuilder.buildCall("Multisig", method)
+        const payload = codec.enc({
+          threshold: multisig.threshold,
+          other_signatories: otherSignatories.map(toSS58),
+          max_weight: weightInfo.weight,
+          maybe_timepoint: multisigInfo?.when,
+          ...(method === "as_multi"
+            ? {
+                call: dynamicBuilder.buildDefinition(lookup.call).dec(callData),
+              }
+            : {
+                call_hash: Binary.fromBytes(callHash),
+              }),
+        })
+        wrappedCallData = mergeUint8(new Uint8Array(location), payload)
+      } catch (_) {
+        throw new Error(
+          `Unsupported runtime version: Multisig.${method} not present or changed substantially`,
+        )
+      }
+
+      return signer.signTx(
+        wrappedCallData,
+        signedExtensions,
+        metadata,
+        atBlockNumber,
+        hasher,
+      )
+    },
+  }
+}
+
+const u8ArrEq = (a: Uint8Array, b: Uint8Array) => {
+  if (a.length != b.length) return false
+  return Array.from(a).every((v, i) => v === b[i])
+}
+
+const cmpU8Arr = (a: Uint8Array, b: Uint8Array) => {
+  for (let i = 0; ; i++) {
+    const overA = i >= a.length
+    const overB = i >= b.length
+
+    if (overA && overB) return 0
+    else if (overA) return -1
+    else if (overB) return 1
+    else if (a[i] !== b[i]) return a[i] > b[i] ? 1 : -1
+  }
+}
```

Comment:
Maybe both of this functions could go to `utils` package. They seem to be quite useful to build future tools.

Reply:
Dunno.... `cmpU8Arr` is only needed for multisig sorting, and u8ArrEq is quite trivial. I wouldn't increase any API surface for this, I think it's best to colocate.

I wondered if `multisig` from substrate bindings should export the utility to sort the signatories, but again... it would just increase the API surface.

Maybe I will though, it could actually be useful for anyone that wants to manually do other_signatories.